### ISSUE TITLE
Distributed A3C with GAE

### ISSUE THREAD

**Issue by 4SkyNet** [deeplearninc/relaax#29]
TODO: Distributed version of A3C with GAE and different neural network structure and optimizer wrt 
https://github.com/4SkyNet/universe-starter-agent

P.S> it's good to make some abstraction for algorithms, which we can build by connecting necessary components such as: 
- neural network structure / weights
- loss function
- optimizer
- input/output filters
- auxiliary functions (from lib)
etc
